File: guess.py
import random

def play_guessing_game():
    number_to_guess = random.randint(1, 100)
    attempts = 0

    print("Welcome to the Guessing Game!")
    print("I have chosen a number between 1 and 100.")

    while True:
        guess = input("Enter your guess: ")
        attempts += 1

        try:
            guess = int(guess)
        except ValueError:
            print("Invalid input. Please enter a valid number.")
            continue

        if guess < number_to_guess:
            print("Too low! Try again.")
        elif guess > number_to_guess:
            print("Too high! Try again.")
        else:
            print(f"Congratulations! You guessed the number in {attempts} attempts.")
            break

play_guessing_game()

File: guess.js
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

function generateRandomNumber(min, max) {
  return Math.floor(Math.random() * (max - min + 1)) + min;
}

function playGuessingGame() {
  const numberToGuess = generateRandomNumber(1, 100);
  let attempts = 0;

  console.log('Welcome to the Guessing Game!');
  console.log('I have chosen a number between 1 and 100.');

  function makeGuess() {
    rl.question('Enter your guess: ', (guess) => {
      attempts++;

      if (isNaN(guess)) {
        console.log('Invalid input. Please enter a valid number.');
        makeGuess();
      } else {
        guess = parseInt(guess);

        if (guess < numberToGuess) {
          console.log('Too low! Try again.');
          makeGuess();
        } else if (guess > numberToGuess) {
          console.log('Too high! Try again.');
          makeGuess();
        } else {
          console.log(`Congratulations! You guessed the number in ${attempts} attempts.`);
          rl.close();
        }
      }
    });
  }

  makeGuess();
}

playGuessingGame();